src/backend/metal/shaders/commands.metal

```diff
@@ -18,18 +18,59 @@ vertex VertexData vs_blit(TextureBlitAttributes in [[stage_in]]) {
     return VertexData { pos, in.src_coords, uint(in.dst_coords.z) };
 }
 
-fragment float4 ps_blit(
+fragment float4 ps_blit_1d(
+    VertexData in [[stage_in]],
+    texture1d<float> tex1D [[ texture(0) ]],
+    sampler sampler2D [[ sampler(0) ]]
+) {
+  return tex1D.sample(sampler2D, in.uv.x);
+}
+
+fragment float4 ps_blit_1d_array(
+    VertexData in [[stage_in]],
+    texture1d_array<float> tex1DArray [[ texture(0) ]],
+    sampler sampler2D [[ sampler(0) ]]
+) {
+  return tex1DArray.sample(sampler2D, in.uv.x, uint(in.uv.z));
+}
+
+fragment float4 ps_blit_2d(
     VertexData in [[stage_in]],
     texture2d<float> tex2D [[ texture(0) ]],
     sampler sampler2D [[ sampler(0) ]]
 ) {
   return tex2D.sample(sampler2D, in.uv.xy, level(in.uv.w));
 }
 
-fragment float4 ps_blit_array(
+fragment float4 ps_blit_2d_array(
     VertexData in [[stage_in]],
     texture2d_array<float> tex2DArray [[ texture(0) ]],
     sampler sampler2D [[ sampler(0) ]]
 ) {
   return tex2DArray.sample(sampler2D, in.uv.xy, uint(in.uv.z), level(in.uv.w));
 }
+
+fragment float4 ps_blit_2d_ms(
```

Comment:
uhm, blit doesn'T support multisampled images

Reply:
heh, ok. I was thinking about GL when adding this. Will remove.